controllers/idle_handlers.go

```diff
@@ -18,16 +18,57 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"os"
 
 	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+// removeDeployment removes the deployment with the specified name
+// in case there is no deployment with the specified name returns false
+func (r *ImageBasedUpgradeReconciler) removeDeployment(osName string) (bool, error) {
+	status, err := r.RpmOSTreeClient.QueryStatus()
```

Comment:
Can we also add a safety check that the booted deployment is not the same stateroot that's being cleaned up? That would be a very bad thing to have happen